File: server/internal/model/trail.go
package model

type Trail struct {
	Id     int      `json:"id" gorm:"primarykey"`
	Name   string   `json:"name"`
	From   string   `json:"from"`
	To     string   `json:"to"`
	Length float64  `json:"length" gorm:"type:decimal(5,2)"`
	Symbol string   `json:"symbol"`
	Point  *Point   `json:"point"`
	Points []*Point `json:"-"`
}

type Point struct {
	Id      int     `json:"-" gorm:"primarykey"`
	TrailId int     `json:"-"`
	Lat     float64 `json:"lat" gorm:"type:decimal(10,8)"`
	Lon     float64 `json:"lon" gorm:"type:decimal(11,8)"`
}

File: mobile/app/src/main/java/com/anonymous/prohiking/data/model/Trail.kt
package com.anonymous.prohiking.data.model

import kotlinx.serialization.Serializable

@Serializable
data class Trail(
    val id: Int,
    val name: String,
    val from: String,
    val to: String,
    val length: Double,
    val symbol: String,
    val point: Point
)

@Serializable
data class Point(
    val lat: Double,
    val lon: Double,
)